### PR TITLE
Xcode 13 conversion

### PR DESCRIPTION
**Pull request by TDH55**
Upgraded `react-natie-track-player` to 2.1.2 in order to make the audio player compatible with Xcode 13. Converted `ProgressBar` to a functional component in order to make the component compatible with the new version of `react-native-track-player` and use the libraries `useProgress` hook

### CODE REVIEW

**Comment by TDH55** [comment]
[Get audio player to work with Xcode 13](https://trello.com/c/gN1FSPvP/323-get-audio-player-to-work-with-xcode-13)

File: .watchmanconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "ignore_dirs": [
+    ".git",
+    "node_modules"
```
Comment (tgalg) [inline .watchmanconfig:4]: looks like this file snuck in from your local testing

**Review by tgalg — COMMENTED** [review]
Curious if this fixes any bugs with audio player we are currently seeing

File: src/components/AudioPlayer/service.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import TrackPlayer from 'react-native-track-player'
 
 // service.js
+//TODO: replace these with hooks
```
Comment (scott-massey) [inline src/components/AudioPlayer/service.js:6]: Do you still need to do this?

File: src/components/AudioPlayer/ProgressBar.js [hunk 3]
```diff
@@ -1,231 +1,212 @@
...
 
-    this.setState({ ending: false })
+    setEnding(false)
   }
 
-  // In the mobile version, played, duration, and progress are all updated by
-  // the render function in ProgressBar, not the AudioPlayerSub, so if there are differences
-  // between the new props and the old props, don't rerender.
-  // Prevents an infinite loop.
-  shouldComponentUpdate(nextProps) {
```
Comment (scott-massey) [inline src/components/AudioPlayer/ProgressBar.js:61]: I'm having trouble seeing it, did you replicate this code to work in the functional component? If not, have you seen any issues during testing?

**Review by scott-massey — COMMENTED** [review]
Generally looks good! Just a couple comments.

File: src/components/AudioPlayer/ProgressBar.js (hunk 3 above)
Comment (TDH55) [reply src/components/AudioPlayer/ProgressBar.js:61]: I haven't come across any issues here yet, I think moving to a functional component made much of what was happening here unecessary

**Review by scott-massey — APPROVED** [review]

**Pull request merged**